### PR TITLE
[impeller] Use mapping based 'Texture::SetContents' in image decoder and text render context

### PR DESCRIPTION
**Pull request by ColdPaleLight**
Fixes the textures portion of https://github.com/flutter/flutter/issues/102262.

cc @chinmaygarde @dnfield 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by chinmaygarde — CHANGES_REQUESTED** [review]
I don't think we should land this. Depending on the backend, a copy may or may not be necessary. And, if a copy is not necessary in the SetContents call but the API only takes mappings, then a caller who cannot transfer ownership of the data into the backend will have no choice but to perform a copy speculatively. If there are both APIs, the caller must prefer mapping but can provide raw pointers. And some backends will be able to avoid copies in those cases as well.

**Comment by ColdPaleLight** [comment]
> I don't think we should land this. Depending on the backend, a copy may or may not be necessary. And, if a copy is not necessary in the SetContents call but the API only takes mappings, then a caller who cannot transfer ownership of the data into the backend will have no choice but to perform a copy speculatively. If there are both APIs, the caller must prefer mapping but can provide raw pointers. And some backends will be able to avoid copies in those cases as well.

I got it, I changed this PR, replace raw pointers with mappings in some places to avoid unnecessary copies, WDYT?

File: impeller/typographer/backends/skia/text_render_context_skia.cc [hunk 1]
```diff
@@ -152,14 +152,14 @@ static std::optional<SkBitmap> CreateAtlasBitmap(const GlyphAtlas& atlas,
...
     std::shared_ptr<Allocator> allocator,
-    const SkBitmap& bitmap,
+    std::shared_ptr<SkBitmap> bitmap,
     size_t atlas_size) {
   TRACE_EVENT0("impeller", __FUNCTION__);
   if (!allocator) {
     return nullptr;
   }
 
-  const auto& pixmap = bitmap.pixmap();
```
Comment (chinmaygarde) [inline impeller/typographer/backends/skia/text_render_context_skia.cc:162]: There is a missing null check here now.

**Review by chinmaygarde — COMMENTED** [review]
This version of the patch makes sense and we should go ahead. I guess the commit message is misleading now though. Can you reword the commit to say something to the effect of "use mapping based SetContexts in image decoder and text render context"?

**Comment by ColdPaleLight** [comment]
> This version of the patch makes sense and we should go ahead. I guess the commit message is misleading now though. Can you reword the commit to say something to the effect of "use mapping based SetContexts in image decoder and text render context"?

Done

File: impeller/typographer/backends/skia/text_render_context_skia.cc (hunk 1 above)
Comment (ColdPaleLight) [reply impeller/typographer/backends/skia/text_render_context_skia.cc:162]: Done. It looks like the `bitmap` here shouldn't be null, so I added a `FML_DCHECK` here.

**Pull request merged**